File: lua/bot.lua
local env = require("redotenv").load()
local discordia = require("discordia")
local client = discordia.Client()

client:on("ready", function()
	print(string.format("Logged in as %s!", client.user.username))
end)

local token = env["TOKEN"]

client:run(string.format("Bot %s", token))

File: elixir/bot.ex
defmodule MyBot do
  use Nostrum.Consumer

  def start_link(_args) do
    Nostrum.Consumer.start_link(__MODULE__)
  end

  def handle_event({:READY, data}) do
    IO.puts("Bot is ready! Logged in as: #{data.user.username}")
    {:ok}
  end

  def handle_event(_event), do: {:ok}
end
